core/src/test/java/org/apache/calcite/avatica/util/ArrayImplTest.java

```diff
@@ -164,6 +165,267 @@
+   *
+   * @throws Exception
+   */
+  @Test
+  public void listOfStructs() throws Exception {
+    // Define the struct type we're creating.
+    ColumnMetaData intMetaData = MetaImpl.columnMetaData("MY_INT", 1, int.class, false);
+    ColumnMetaData stringMetaData = MetaImpl.columnMetaData("MY_STRING", 2, String.class, true);
+    StructType structType = ColumnMetaData.struct(Arrays.asList(intMetaData, stringMetaData));
+    // Create some structs.
+    Struct struct1 = new StructImpl(Arrays.<Object>asList(1, "one"));
+    Struct struct2 = new StructImpl(Arrays.<Object>asList(2, "two"));
+    Struct struct3 = new StructImpl(Arrays.<Object>asList(3));
+    Struct struct4 = new StructImpl(Arrays.<Object>asList(4, "four"));
+    ArrayType arrayType = ColumnMetaData.array(structType, "OBJECT", Rep.STRUCT);
+    ColumnMetaData arrayMetaData = MetaImpl.columnMetaData("MY_ARRAY", 1, arrayType, false);
+    ArrayImpl.Factory factory = new ArrayFactoryImpl(Unsafe.localCalendar().getTimeZone());
+
+    // Create some lists from the structs, instead of an array.
+    List<Object> list1 = new ArrayList<Object>(2);
+    list1.add(struct1);
+    list1.add(struct2);
+    List<Object> list2 = new ArrayList<Object>(2);
+    list2.add(struct3);
+    list2.add(struct4);
+    List<List<Object>> rows = Arrays.asList(Collections.<Object>singletonList(list1),
+        Collections.<Object>singletonList(list2));
+
+    // Create two rows, each with one (array) column.
+    try (Cursor cursor = new ListIteratorCursor(rows.iterator())) {
+      List<Accessor> accessors = cursor.createAccessors(Collections.singletonList(arrayMetaData),
+          Unsafe.localCalendar(), factory);
+      assertEquals(1, accessors.size());
+      Accessor accessor = accessors.get(0);
+
+      assertTrue(cursor.next());
+      Array actualArray = accessor.getArray();
+
+      Object[] arrayData = (Object[]) actualArray.getArray();
+      assertEquals(2, arrayData.length);
+      Struct actualStruct = (Struct) arrayData[0];
+      Object[] o = actualStruct.getAttributes();
+      assertEquals(2, o.length);
+      assertEquals(1, o[0]);
+      assertEquals("one", o[1]);
+
+      actualStruct = (Struct) arrayData[1];
+      o = actualStruct.getAttributes();
+      assertEquals(2, o.length);
+      assertEquals(2, o[0]);
+      assertEquals("two", o[1]);
+
+      assertTrue(cursor.next());
+      actualArray = accessor.getArray();
+      arrayData = (Object[]) actualArray.getArray();
+      assertEquals(2, arrayData.length);
+      actualStruct = (Struct) arrayData[0];
+      o = actualStruct.getAttributes();
+      assertEquals(1, o.length);
+      assertEquals(3, o[0]);
+
+      actualStruct = (Struct) arrayData[1];
+      o = actualStruct.getAttributes();
+      assertEquals(2, o.length);
+      assertEquals(4, o[0]);
+      assertEquals("four", o[1]);
+    }
+  }
+
+  /**
+   * Plain Java object class for the two tests that follow.
```

Comment:
The comment does not seem to add much value, I think it can be safely removed

Reply:
:core:checkstyleTest would fail with error "Missing a Javadoc comment" if we remove that docblock.